### PR TITLE
Migrate eslint config edx

### PR DESCRIPTION
**Pull request by abdullahwaheed**
### Ticket
[Migrate eslint-config-edx](https://github.com/edx/upgrades/issues/79) 

### What has changed
Migrated `eslint-config-edx` to `@edx/eslint-config` and fixed related issues
Fixed startup issues and updated Dockerfile accordingly
Updated some legacy code
Added dev server startup script

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @abdullahwaheed! I've created [OSPR-6755](https://openedx.atlassian.net/browse/OSPR-6755) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

**Comment by natabene** [comment]
@abdullahwaheed Thank you for the contribution! Once you have filled out the forms and they have been processed, we will be able to review this and all your future code contributions.

**Comment by natabene** [comment]
@abdullahwaheed I was informed that you are part of the new squad. Please work with @jmbowman to ensure you are added to the contributors list with tCRIL before this is merged.

**Review by davidjoy — APPROVED** [review]
One comment about dependencies - otherwise I don't see any red flags.  We should test the code changes before merging to ensure we didn't cause any regressions, but this looks like a fine upgrade.

File: package.json
Comment (abdullahwaheed) [reply package.json:20]: let me move `enzyme` to devDependency. not sure about `acorn` i think it might be a peerDependency. Will remove its unnecessary

**Comment by davidjoy** [comment]
To be clear - the FED-BOM team should feel free to give this a test in studio-frontend and merge once you're satisfied the affected functionality still works, if that hasn't happened already!  As a representative from "frontend triage", I looked through and didn't find anything that I thought was concerning - this seems like a fine upgrade - but I haven't tested the code changes myself.  It might be a good idea to have a more detailed review from the FED-BOM team itself.

**Comment by abdullahwaheed** [comment]
> To be clear - the FED-BOM team should feel free to give this a test in studio-frontend and merge once you're satisfied the affected functionality still works, if that hasn't happened already! As a representative from "frontend triage", I looked through and didn't find anything that I thought was concerning - this seems like a fine upgrade - but I haven't tested the code changes myself. It might be a good idea to have a more detailed review from the FED-BOM team itself.

@davidjoy tested it with some of the main pages. I would recommend that owning team should test it on sandbox. Working fine for me

**Review by rayzhou-bit — APPROVED** [review]
This looks great! Since you tested it as well, I think it's ready for merge.

**Comment by abdullahwaheed** [comment]
@davidjoy upon merging this PR, Github pages deployment script is [failing](https://github.com/openedx/studio-frontend/runs/7514605961?check_suite_focus=true). Does package `deploy-storybook-ci` outdate and we have to use some alternate?

**Pull request merged**
